### PR TITLE
Fix QAOA

### PR DESCRIPTION
**Pull request by t-imamichi**
### Summary

Fix QAOA of opflow.

### Details and comments

- Allow users to change the number of qubits of QAOA
  - Change the order of check and varform initialization 
- Track appropriate parameters associated with QAOA var form
  - Add a check of number of qubits

### CODE REVIEW

**Comment by t-imamichi** [comment]
I found that this change breaks some unit tests of chemistry, e.g., test/chemistry/test_uccsd_hartree_fock.py
I need to check the details.

File: qiskit/algorithms/minimum_eigen_solvers/vqe.py [hunk 1]
```diff
@@ -300,6 +300,7 @@ def construct_expectation(self,
 
         observable_meas = self.expectation.convert(StateFn(operator, is_measurement=True))
         ansatz_circuit_op = CircuitStateFn(wave_function)
+        self._var_form_params = sorted(ansatz_circuit_op.parameters, key=lambda p: p.name)
```
Comment (woodsp-ibm) [inline qiskit/algorithms/minimum_eigen_solvers/vqe.py:303]: As the var_form_params are set sorted by the check_operator method, which is called ahead of this, in this same method, is this needed here, or just duplicates what was just done again? Neither the var form nor the operator should be None at this point I think whereby the sorted line in check_operator would be skipped.

Comment (t-imamichi) [reply qiskit/algorithms/minimum_eigen_solvers/vqe.py]: Sorry. I found out that this change is wrong. The root cause was the missing comparison of number of qubits in QAOA's _check_operator.

**Comment by woodsp-ibm** [comment]
Ok. At first sight it looks like the order of the check_operator and creation of var form were just swapped with the added qubit check that you would think was for efficiency to prevent building out the var form again, and if you did it would not be an issue. The check though is in fact needed because of this parameter sorting.

With the code change on the move from Aqua, which removed the operator from constructor and the setter, meaning it can only be passed on the compute_min_eigval, I think going forwards there is room to improve refactor/improve the logic around this checking/building etc. Especially if the parameters end up with a defined ordering #5614

**Review by woodsp-ibm — APPROVED** [review]

**Comment by t-imamichi** [comment]
CI fails due to HTTP request timed out of pip. How can I rerun it?

**Pull request merged**